lms/services/mailchimp.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from dataclasses import dataclass
+
+import mailchimp_transactional
+
+LOG = logging.getLogger(__name__)
+
+
+@dataclass
+class EmailSender:
+    subaccount: str
+    """The Mailchimp subaccount ID to use to send the email."""
+
+    email: str
+    """The email address that the email will appear to come from."""
+
+    name: str
+    """The full name that the email will appear to come from."""
+
+
+@dataclass
+class EmailRecipient:
+    email: str
+    """The email address to send to."""
+
+    name: str
+    """The recipient full name to use in the email's To: header."""
+
+
+class MailchimpService:
+    def __init__(self, api_key):
+        if api_key:
+            self.mailchimp_client = mailchimp_transactional.Client(api_key)
+
+    def send_template(
+        self,
+        template_name: str,
+        sender: EmailSender,
+        recipient: EmailRecipient,
+        template_vars: dict,
+    ):
+        """
+        Send an email using Mailchimp Transactional's send-template API.
+
+        https://mailchimp.com/developer/transactional/api/messages/send-using-message-template/
+        """
+        params = {
+            "template_name": template_name,
+            "message": {
+                "subaccount": sender.subaccount,
+                "from_email": sender.email,
+                "from_name": sender.name,
+                "to": [{"email": recipient.email, "name": recipient.name}],
+                "track_opens": True,
+                "track_clicks": True,
+                "global_merge_vars": [
+                    {"name": key, "content": value}
+                    for key, value in template_vars.items()
+                ],
+            },
+            "async": True,
```

Comment:
I think there's definitely going to be some learning here once we start monitoring this in actual usage. For example right now (with `async` or not) any one failure from the Mailchimp API will cause us not to even try to send any further emails in the rest of the batch.

I think we probably do want to use `async` because it's the Mailchimp API's "bulk sending mode" and bulk sending is what we're doing here.

Although I think it's true that we don't know whether or not we actually need it, we may be able to get away without `async`.

I don't have any plans to do anything with any rejections or errors that we might get from non-`async` Mailchimp API calls anyway, at least not for this first version, so I don't think we're losing anything.